zhaquirks/quirky/tapt.py

```diff
@@ -0,0 +1,187 @@
+"""GE Quirky TAPT device."""
+import logging
+
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+import zigpy.types as t
+from zigpy.zcl.clusters.general import (
+    Basic,
+    Groups,
+    Identify,
+    LevelControl,
+    OnOff,
+    Ota,
+    Scenes,
+)
+from . import QUIRKY
+from ..const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+    SHORT_PRESS,
+    COMMAND_ON,
+    TURN_ON,
+    TURN_OFF,
+    LONG_PRESS,
+    DIM_UP,
+    DIM_DOWN,
+    LONG_RELEASE,
+    BUTTON,
+    COMMAND,
+    COMMAND_OFF,
+    COMMAND_MOVE,
+    COMMAND_STOP,
+    CLUSTER_ID,
+    ENDPOINT_ID,
+    ARGS,
+)
+
+TAPT_MFG_CODE = 0x5488  # 21640 base 10
+TAPT_NODE_MFG_CODE = 0xBBAA  # 48042 base 10
+TAPT_CLUSTER = 0xFC20  # 64544 base 10
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class TaptConfig(CustomCluster):
+    """TAPT Configuration Cluster."""
+
+    cluster_id = TAPT_CLUSTER
+    name = "TAPTCluster"
+    ep_attribute = "tapt_cluster"
+    server_commands = {}
+    client_commands = {}
+    attributes = {
+        0x0000: ("Switch Mode", t.uint8_t),
+        # Value, Description
+        # 0 - Standard On/Off Switch, No Smart Status, Relay Status Endpoint 2
+        # 1 - Standard On/Off Switch, Smart Status on Endpoint 1, Relay Status on Endpoint 2
+        # 2 - Smart Switch, Mains always On, Smart Status on Endpoint 1
+        # 3 - Upper Toggles Load, Smart Status on Endpoint 1, Relay Status on Endpoint 2
+        # 4 - Lower Toggles Load, Smart Status on Endpoint 1, Relay Status on Endpoint 2
+    }
+
+    async def read_attributes_raw(self, attributes, manufacturer=None):
```

Comment:
Why not just replace none in the signatures with the mfg code you want? @Adminiuga thoughts?

Reply:
I think we do that in HA channels, not in the clusters. But I was thinking lately to do it in zigpy instead. Then you could override the mfg code in the quirks

@coolacid what is the mfg code reported by the node descriptor?

Reply:
When read_attributes_raw or write_attributes fires while using the Get/Set attributes in the HA UI, it uses the mfg code from the node descriptor, it doesn't appear to ever be none. 

I went up and down the classes trying to find a place I could just override the mfg code as a simple setting for the custom cluster, however I couldn't find any, thus resorting to overloading these two functions - This allowed me to Get/Set the attributes without having to specify the mfg code in the UI every time - To me, that's a quality of life bit for people that want to make use of this setting. 

I actually define, and check for the mfg code from the node descriptor, just in case somehow this custom cluster gets added to something else - but as far as I can tell, this is the only device under the quirky brand - TAPT_NODE_MFG_CODE = 0xBBAA  # 48042 base 10

Reply:
@Adminiuga what are your thoughts on how this was done? I saw some mfg specific PR’s to zigpy iirc. Were they related to this sort of thing?

Reply:
fixed in https://github.com/zigpy/zigpy/pull/429 see PR description for details